mypy/config_parser.py

```diff
@@ -39,7 +39,12 @@ def expand_path(path: str) -> str:
     the provided path.
     """
 
-    return os.path.expandvars(os.path.expanduser(path))
+    expanded_path = os.path.expandvars(os.path.expanduser(path))
+    if 'MYPY_CONFIG_FILE_DIR' not in os.environ:
+        os.environ['MYPY_CONFIG_FILE_DIR'] = expanded_path
+    else:
+        os.environ['MYPY_CONFIG_FILE_DIR'] += (os.pathsep + expanded_path)
+    return expanded_path
```

Comment:
That looks wrong to me: `expand_path` appears to be a generic method.
It should probably be done here instead: https://github.com/python/mypy/pull/8574/files#diff-09533a8b2bebadfae430681eb15b01c9R120 - i.e. setting it to the current file before parsing it.

Reply:
The link you gave above points to changed code itself. I don't understand your comment. Please, can you elaborate about where should the code be changed ?

Reply:
@ashutosh1919 
Needs to be expanded via clicking, sorry.
https://github.com/python/mypy/blob/00e8024ffd187bdbddeb71fcced53cd2912a9410/mypy/config_parser.py#L120
But that is just a guess.
The idea would be to define it before parsing a config file, then it gets / can be expanded.

Reply:
@blueyed , Please take a look at it now.